### ISSUE TITLE
IntelliSense doesn't work in watch window

### ISSUE THREAD

**Issue by CreateAndInject** [dotnet/roslyn#42718]
**Version Used**: 16.5

**Steps to Reproduce**:

1. Write following code
```
        static void Main(string[] args)
        {
            for (int num = 0; num < 10; num++)
                Console.WriteLine(num);
        }
```
2. Debug and break in `Console.WriteLine(num);`
3. Type `num` in watch window

**Expected Behavior**: IntelliSense show `num` in the context menu.

**Actual Behavior**: There's no `num` in context menu
![image](https://user-images.githubusercontent.com/2529999/77408679-0e24e180-6df3-11ea-846c-da134686c093.png)

**Comment by genlu** [comment]
So the syntax tree passed to completion provider looked like this. Basically, we are missing the bracket which would have includeed the cursor inside the loop context. @tmat Do you know how the new tree is constructed when using watch window? Is this in EE?

```cs
        static void Main(string[] args)
        {
            int x = 10;
            for (int num = 0; num < 10; num++)
                Console.WriteLine(num);
;s;        }
```

**Comment by tmat** [comment]
No, this is IDE. I don't know how it really works. It's a big hack :(

**Comment by genlu** [comment]
hmm, @CyrusNajmabadi @jasonmalinowski do you guys know how this works?

**Comment by jasonmalinowski** [comment]
In a word:

![image](https://user-images.githubusercontent.com/201340/78618689-1d168580-7830-11ea-85f4-3ffefb2654c7.png)

In a few more words, we essentially are creating a projection buffer which merges the contents of the surrounding file and the watch expression to stitch them together as 'real' code and from there we can run queries. You probably want to start debugging under https://github.com/dotnet/roslyn/blob/e944d821bc912aac99edfad84d820b549218a596/src/VisualStudio/CSharp/Impl/LanguageService/CSharpDebuggerIntelliSenseContext.cs and see where that takes you.

Good luck.

**Comment by genlu** [comment]
There's a VS feedback on this issue:
https://developercommunity2.visualstudio.com/t/Intellisense-not-showing-after-dot-withi/1248527?preview=true

**Comment by genlu** [comment]
While we are fixing the issue, you can rewrite the code with a brace enclosed body as a quick workaround. For example

from 
```cs
for (int num = 0; num < 10; num++)
    Console.WriteLine(num);
```
to
```cs
for (int num = 0; num < 10; num++)
{
    Console.WriteLine(num);
}
```

from 
```cs
.Where(t => 
                   t.Namespace == "somenamespace")
```
to
```cs
.Where(t => 
            {return  t.Namespace == "somenamespace";})
```

**Comment by CreateAndInject** [comment]
This issue is marked as milestone 16.9, but 16.9 doesn't fix it.

**Comment by genlu** [comment]
I have moved it to backlog for now, since I don't have capacity to investigate it at the moment and there's a workaround for this issue as I mentioned above. @vatsalyaagrawal Let me know if this should be in 16.10.
